nevergrad/functions/ml/mlfunctionlib.py

```diff
@@ -173,11 +167,17 @@ def __init__(
         # For the evaluation we remove the noise (unless overfitter)
         evalparams["noise_free"] = not overfitter
         super().__init__(partial(self._ml_parametrization, **params), parametrization.set_name(""))
-        self.evaluation_function = partial(self._ml_parametrization, **evalparams)  # type: ignore
+        self._evalparams = evalparams
         self.register_initialization(regressor=regressor, data_dimension=data_dimension, dataset=dataset,
                                      overfitter=overfitter)
 
-    def get_dataset(self, data_dimension: tp.Optional[int], dataset: str) -> None:
+    def evaluation_function(self, *args: tp.Any, **kwargs: tp.Any) -> float:
+        assert not args
+        # override with eval parameters (with partial, the eval parameters would be overriden by kwargs)
+        kwargs.update(self._evalparams)
+        return self._ml_parametrization(**kwargs)
```

Comment:
@teytaud with partial, the parametrization would override the evaluation parameters, which I expect was not the expected behavior

Reply:
Enormous thx.